modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java

```diff
@@ -63,343 +54,272 @@
+
+	public void init(SynapseEnvironment se) {
+		this.synapseEnvironment = se;
+		initMessageSender(parameters);
+		if (!isPinnedServer(se.getServerContextInformation().getServerConfigurationInformation()
+		                      .getServerName())) {
+			/*
+			 * If it is not a pinned server we do not start the message
+			 * processor. In that server.
+			 */
+			setActivated(false);
+		}
+		super.init(se);
+		/*
+		 * initialize the task manager only once to alleviate complexities
+		 * related to the pending tasks.
+		 */
+		if (taskManager == null) {
+			taskManager = synapseEnvironment.getSynapseConfiguration().getTaskManager();
+		}
+
+		/*
+		 * If the task manager is not initialized yet, subscribe to
+		 * initialization completion event here.
+		 */
+		if (!taskManager.isInitialized()) {
+			taskManager.addObserver(this);
+			return;
+		}
+
+		if (Boolean.parseBoolean(String.valueOf(parameters.get(MessageProcessorConstants.IS_ACTIVATED))) &&
+		    !isDeactivated()) {
+			this.start();
+		}
+
+	}
+
+	public boolean start() {
+		for (int i = 0; i < memberCount; i++) {
+			/*
+			 * Make sure to fetch the task after initializing the message sender
+			 * and consumer properly. Otherwise you may get NullPointer
+			 * exceptions.
+			 */
+			Task task = this.getTask();
+			TaskDescription taskDescription = new TaskDescription();
+			taskDescription.setName(TASK_PREFIX + name + i);
+			taskDescription.setTaskGroup(MessageProcessorConstants.SCHEDULED_MESSAGE_PROCESSOR_GROUP);
+			/*
+			 * If this interval value is less than 1000 ms, ntask will throw an
+			 * exception while building the task. So to get around that we are
+			 * setting threshold interval value of 1000 ms to the task
+			 * description here. But actual interval value may be less than 1000
+			 * ms, and hence isThrotling is set to TRUE.
+			 */
+			if (interval < MessageProcessorConstants.THRESHOULD_INTERVAL) {
+				taskDescription.setInterval(MessageProcessorConstants.THRESHOULD_INTERVAL);
+			} else {
+				taskDescription.setInterval(interval);
+			}
+			taskDescription.setIntervalInMs(true);
+			taskDescription.addResource(TaskDescription.INSTANCE, task);
+			taskDescription.addResource(TaskDescription.CLASSNAME, task.getClass().getName());
+
+			/*
+			 * If there is a Cron Expression we need to set it into the
+			 * TaskDescription so that the framework will take care of it.
+			 */
+			if (cronExpression != null) {
+				taskDescription.setCronExpression(cronExpression);
+			}
+			taskManager.schedule(taskDescription);
+		}
+		logger.info("Started message processor. [" + getName() + "].");
+
+		return true;
+	}
+
+	public boolean isDeactivated() {
+		return taskManager.isTaskDeactivated(TASK_PREFIX + name + DEFAULT_TASK_SUFFIX);
+	}
+
+	public void setParameters(Map<String, Object> parameters) {
+		super.setParameters(parameters);
+
+		if (parameters != null && !parameters.isEmpty()) {
+			Object o = parameters.get(MessageProcessorConstants.CRON_EXPRESSION);
```

Comment:
Instead of saying Object o, can we have a meaningful name for the object reference.